optax/_src/complex_valued.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2021 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Complex-valued optimization.
+
+When using `split_real_and_imaginary` to wrap an optimizer, we split the complex
+parameters and updates into pairs of real ones before sending them to the
+`update` of the wrapped optimizer, and merge the pairs of transformed real
+updates into complex ones afterward. In this way, optimizers on complex
+parameters behave the same way as if they were running on two real parameters.
+
+Note that the convention of conjugate for complex gradients in JAX is different
+from that in PyTorch and other frameworks, and we need to manually conjugate the
+gradients between `jax.grad` and `optimizer.update`.
+
+See details at https://github.com/deepmind/optax/issues/196
```

Comment:
(Missing a full stop at the end here).

Reply:
Do we really need a full stop after a URL? I think that it will cause confusion when copying or rendering the URL.

Should it be like this?
```
See details at https://github.com/deepmind/optax/issues/196.
```
Or we add a space like this?
```
See details at https://github.com/deepmind/optax/issues/196 .
```